tests/integration/cli_test.py

```diff
@@ -98,20 +98,56 @@ def test_pull_with_digest(self, mock_logging):
             'sha256:38a203e1986cf79639cfb9b2e1d6e773de84002feea2d4eb006b52004ee8502d)...')
 
     @mock.patch('sys.stdout', new_callable=StringIO)
-    def test_build_no_cache(self, mock_stdout):
+    def test_build_plain(self, mock_stdout):
         self.command.base_dir = 'tests/fixtures/simple-dockerfile'
         self.command.dispatch(['build', 'simple'], None)
 
         mock_stdout.truncate(0)
         cache_indicator = 'Using cache'
+        pull_indicator = 'Status: Image is up to date for busybox:latest'
         self.command.dispatch(['build', 'simple'], None)
         output = mock_stdout.getvalue()
         self.assertIn(cache_indicator, output)
+        self.assertNotIn(pull_indicator, output)
+
+    @mock.patch('sys.stdout', new_callable=StringIO)
```

Comment:
As an aside, I think it's a mistake we do this patch as a decorator.  It makes it harder to debug tests with `pdb`.  I've been moving some of them to be context managers around just the single line that needs the patch.

Using the `with mock.patch()` around just the line that needs it also removes the need to do `mock_stdout.truncate(0)`, since it would only wrap the stdout we care about.


Since this is pretty common in our test suite, I wouldn't ask you to change it as part of this PR, I think we can address that in a separate cleanup branch and change them all at once.